webapp/lib/view.dart

```diff
@@ -227,6 +228,77 @@ class ProjectSelectorView {
   }
 }
 
+class ChartFiltersView {
+  static final ChartFiltersView _singleton = new ChartFiltersView._internal();
+
+  factory ChartFiltersView() {
+    return _singleton;
+  }
+
+  DivElement chartFiltersContainer;
+  DivElement _singleFilterSpan;
+  LabelElement _periodFilterTitle;
+  SelectElement _periodFilter;
+
+  ChartFiltersView._internal() {
+    chartFiltersContainer = new DivElement()..classes.add('chart-filters');
+    _singleFilterSpan = new DivElement()..classes.add('chart-filter');
+    _periodFilterTitle = new LabelElement()..text = 'Filter by Period:';
+    _periodFilter = new SelectElement()..classes.add('period-filter');
+    _periodFilter.children.addAll(_getPeriodFilterOptions());
+    _periodFilter.onChange.listen(filterChartsByPeriod);
+    _singleFilterSpan.append(_periodFilterTitle);
+    _singleFilterSpan.append(_periodFilter);
+    chartFiltersContainer.append(_singleFilterSpan);
+  }
+
+  String get selectedPeriodFilter => _periodFilter.value;
+
+  void filterChartsByPeriod(_) {
+    var periodFilter = controller.ChartPeriodFilters.values.firstWhere((filter) => filter.toString() == selectedPeriodFilter);
+    if (periodFilter != null) {
+      var filterData = new controller.ChartFilterdata()..periodFilter = periodFilter;
+      controller.command(controller.UIAction.needsReplyDataUpdated, filterData);
+      controller.command(controller.UIAction.systemEventsDataUpdated, filterData);
+    }
+  }
+
+  List<OptionElement> _getPeriodFilterOptions() {
+    List<OptionElement> periodsOptions = [];
+    var defaultOption = new OptionElement()
```

Comment:
I'd rather not have a default options that doesn't show anything sensible on the screen. As per my other comment, let's turn this into the option to show data for all time.

Reply:
Implemented an `allTime` option as per your earlier comment